src/qutip_qip/compiler/gatecompiler.py

```diff
@@ -80,8 +92,11 @@ def __init__(self, num_qubits=None, params=None, pulse_dict=None, N=None):
             "GLOBALPHASE": self.globalphase_compiler,
             "IDLE": self.idle_compiler
         }
-        self.args = {}
-        self.args.update({"params": self.params})
+        self.args = {  # Default configuration
+            "shape": "rectangular",
+            "num_samples": None,
+            "params": self.params,
```

Comment:
I think we should think about whether `shape`, `num_samples` and `params` belong in `self.args` or not. Quite possibly how it is is currently fine, but we should have a clear idea of why some attributes are in `args` and others not.

Reply:
If `args` is "what is passed to the compiler functions", then I think these are fine here.